### PR TITLE
Improve tests and static checks to avoid flaky pipelines

### PR DESCRIPTION
**Pull request by tranchitella**

### CODE REVIEW

**Comment by tranchitella** [comment]
I won't run the pipeline, as there are no changes in the code.

**Review by lluiscampos — APPROVED** [review]

File: tests/tests/test_connect.py [hunk 1]
```diff
@@ -151,6 +151,12 @@ def test_connect(self, clean_mongo, tenant_id=None):
                     "body": b"sh-5.0$ ",
                 }
 
+        for i in range(15):
+            obj = api_mgmt.get_device(dev.id)
+            if obj.status != 'connected':
```
Comment (alfrunes) [inline tests/tests/test_connect.py:156]:
```suggestion
            if obj.status == 'connected':
```

Comment (tranchitella) [reply tests/tests/test_connect.py:156]: @alfrunes ... why?

Comment (alfrunes) [reply tests/tests/test_connect.py:156]: Shouldn't we break the loop once the device has connected?

Comment (tranchitella) [reply tests/tests/test_connect.py:156]: here we are waiting for the disconnection of the device; the next step (a few lines below) is checking we get 404

Comment (alfrunes) [reply tests/tests/test_connect.py:156]: Ohh, I just assumed it was the other way around. Sorry...

**Pull request merged**